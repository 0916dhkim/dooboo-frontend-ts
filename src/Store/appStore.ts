import { observable, action } from 'mobx';
import autobind from 'autobind-decorator';

@autobind
class Store {

  @observable grey: boolean = false;

  appColorChange() {
    this.grey = !this.grey;
  }
}

export default Store;
