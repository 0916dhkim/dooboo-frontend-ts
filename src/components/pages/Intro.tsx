import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { inject, observer } from 'mobx-react';
import autobind from 'autobind-decorator';
import Radium from 'radium';

import { globalColors, globalEffects } from '@utils/styles';
import CooniBtn from '@shared/CooniBtn';

@autobind
@inject('store')
@observer
@Radium
class Intro extends Component<any> {

  public login() {
    this.props.store.logIn();
  }

  public signUp() {
    this.props.history.push('/signup');
  }

  public render() {
    return(
      <div>
        {
          this.props.store.loggedIn
          ? <Redirect to='/tab/tab1' />
          : <div style={styles.container}>
              <div style={styles.introContainer}>
                <div style={styles.oauthBox}>
                    <CooniBtn
                      clickHandler={() => this.signUp()}
                      white={true}
                      btnTxt='회원가입'
                    />
                    <CooniBtn
                      clickHandler={() => this.login()}
                      white={true}
                      btnTxt='구글으로 로그인'
                      imgSrc='https://icdn6.digitaltrends.com/image/google_icon-377x372.jpg'
                    />
                    <CooniBtn
                      clickHandler={() => this.login()}
                      white={true}
                      btnTxt='페이스북으로 로그인'
                      // tslint:disable-next-line:max-line-length
                      imgSrc='https://media4.s-nbcnews.com/j/streams/2013/december/131203/2d9840012-facebooklogo.nbcnews-fp-360-360.jpg'
                    />
                </div>
              </div>
            </div>
        }
      </div>
    );
  }
}

const styles: any = {
  container: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    background: globalColors.cooniGradient,
  },

  introContainer: {
    margin: '28px',
    width: 'calc(100%-56px)',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    position: 'relative',
  },

  oauthBox: {
    bottom: 0,
    position: 'absolute',
  },

  oauthBtn: {
    color: '#fff',
    width: '320px',
    height: '28px',
    borderRadius: '3px',
    border: 'solid 2px #fff',
    margin: '8px 0',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',

    ...globalEffects.btnEffect,

    ':hover': {
      ...globalEffects.btnEffectHover,
    },
    ':active': {
      opacity: 0.5,
    },

    '@media (max-width: 740px)': {
      width: '85vw',
      height: '8vh',
    },
  },

  googleLogoImg: {
    position: 'absolute',
    left: '16px',
    height: '20px',
    width: '20px',
    objectFit: 'cover',
  },

  fbLogoImg: {
    position: 'absolute',
    left: '16px',
    height: '20px',
    width: '20px',
    objectFit: 'cover',
  },
};

export default Intro;
