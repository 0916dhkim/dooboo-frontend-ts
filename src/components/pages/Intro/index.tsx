import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { inject, observer } from 'mobx-react';

import { colors, effects } from '@utils/styles';
import CooniBtn from '@shared/CooniBtn';
import {
  google_logo_1x, google_logo_2x, google_logo_3x,
  facebook_logo_1x, facebook_logo_2x, facebook_logo_3x,
} from '@utils/Icons';

const styles = require('./style.css');

@inject('store')
@observer
class Intro extends Component<any> {
  public login(oauth: string) {
    this.props.store.user.login(`user from ${oauth}`);
  }

  public goToSignUp() {
    this.props.history.push('/signup');
  }

  public render() {
    const { getString } = this.props.store.locale;
    return(
      <div>
        {
          this.props.store.user.loggedIn
          ? <Redirect to='/tab/tab1' />
          : <div className={styles.container}>
              <div className={styles.introContainer}>
                <div className={styles.oauthBox}>
                    <CooniBtn
                      onClick={() => this.goToSignUp()}
                      white={true}
                      btnTxt={getString('SIGNUP')}
                    />
                    <CooniBtn
                      onClick={() => this.login('google')}
                      white={true}
                      btnTxt={getString('GOOGLE_LOGIN')}
                      imgSrc={google_logo_1x}
                      srcset={`${google_logo_1x} 1x, ${google_logo_2x} 2x, ${google_logo_3x} 3x`}
                    />
                    <CooniBtn
                      onClick={() => this.login('facebook')}
                      white={true}
                      btnTxt={getString('FACEBOOK_LOGIN')}
                      imgSrc={facebook_logo_2x}
                      srcset={`${facebook_logo_1x} 1x, ${facebook_logo_2x} 2x, ${facebook_logo_3x} 3x`}
                    />
                </div>
              </div>
            </div>
        }
      </div>
    );
  }
}

export default Intro;
