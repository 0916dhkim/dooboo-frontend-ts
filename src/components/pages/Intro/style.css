:root {
  --cooniGradient: linear-gradient(to bottom right, rgb(12, 157, 197), rgb(201, 109, 216));
  --cooniGradientHorizontal: linear-gradient(to right, rgb(12, 157, 197), rgb(201, 109, 216));
  --cooniTxtColor: rgb(13, 157, 197);
  --cooniGrey: rgb(227, 227, 227);
  --cooniBtnGrey: rgba(0, 0, 0, 0.4);
}

/* postcss will autoprefix the css properties */
.container {
  display: flex;
  background: var(--cooniGradient);
  position: absolute;
  width: 100%;
  height: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.introContainer {
  margin: 28px;
  width: calc(100%-56px);
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
}

.oauthBox {
  bottom: 0;
  position: absolute;
};

.oauthBtn {
  color: #fff;
  width: 320px;
  height: 28px;
  border-radius: 3px;
  border: solid 2px #fff;
  margin: 8px 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: 'pointer';
  opacity: 1;
  transition: 'opacity 0.2s ease-in-out';

  &:hover {
    opacity: 0.5;
  }

  &:active {
    opacity: 0.5;
  };

  @media (max-width: 740px) {
    width: 85vw;
    height: 8vh;
  }
}

.googleLogoImg {
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover;
}

.fbLogoImg {
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover;
}