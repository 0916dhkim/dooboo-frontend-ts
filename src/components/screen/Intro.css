@import '../../root.css';

/* postcss will autoprefix the css properties */
.container {
  height: 100%;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.background {
  background: var(--cooniGradient);
}

.oauthBox {
  bottom: 0;
  position: absolute;
};
