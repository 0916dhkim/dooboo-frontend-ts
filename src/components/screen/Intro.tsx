import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import styled from 'styled-components';

import Button from '../shared/Button';

import { inject, observer } from 'mobx-react';
import { media } from '../../theme';
import Store from 'src/stores/appStore';

const Container = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  align-self: stretch;
  overflow: scroll;
  background: linear-gradient(to bottom right, rgb(12, 157, 197), rgb(201, 109, 216));

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const Box = styled.div`
  position: absolute;
  width: 85vw;
  bottom: 40px;

  ${media.mobile`
    width: 50vw;
  `}
`;

interface IProps {
  store?: Store;
}

@inject('store') @observer
class Intro extends Component<IProps, {}> {
  public render() {
    const { getString } = this.props.store.locale;
    return(
      <Container>
        <Box>
          <Button
            id='btn'
            onClick={() => this.onClick()}
            white={true}
            txt={getString('BUTTON')}
          />
        </Box>
      </Container>
    );
  }

  private onClick = () => {
    alert('clicked');
  }
}

export default Intro;
