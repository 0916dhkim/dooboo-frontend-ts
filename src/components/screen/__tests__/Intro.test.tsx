import React from 'react';
import { shallow, render } from 'enzyme';

import Store from '../../../stores/appStore';
import { Intro } from '../Intro';

// test for the container page in dom
describe('Intro page DOM rendering test', () => {
  const testStore = new Store();
  const page = shallow(
    <Intro store={testStore} />,
  );

  it('Tab1 page has to be matched to snapshot', () => {
    expect(page).toMatchSnapshot();
  });
});
