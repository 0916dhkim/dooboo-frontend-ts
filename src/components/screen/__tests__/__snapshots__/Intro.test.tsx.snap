// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<div
  className="sc-AykKH kEvPvw"
>
  <div
    className="sc-AykKI jvsToU"
  >
    <span
      className="sc-AykKK ztLDi"
    >
      
    </span>
    <span
      className="sc-AykKK ztLDi"
    >
      
    </span>
    <span
      className="sc-AykKK ztLDi"
    >
      
    </span>
  </div>
  <div
    className="sc-AykKJ dbMAGP"
  >
    <button
      className="sc-AykKC hXMJWl"
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "height": "60px",
        }
      }
    >
      <div
        className="sc-AykKD eXbViw"
      >
        <img
          className="sc-AykKF eeCYSW"
          src="test-file-stub"
        />
        <span
          className="sc-AykKE ekRPzN"
        >
          Login
        </span>
      </div>
    </button>
    <button
      className="sc-AykKC hXMJWl"
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "height": "60px",
        }
      }
    >
      <div
        className="sc-AykKD eXbViw"
      >
        <span
          className="sc-AykKE ekRPzN"
        >
          Navigate
        </span>
      </div>
    </button>
    <button
      className="sc-AykKC hXMJWl"
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "height": "60px",
        }
      }
    >
      <div
        className="sc-AykKD eXbViw"
      >
        <span
          className="sc-AykKE ekRPzN"
        >
          Change theme
        </span>
      </div>
    </button>
  </div>
</div>
`;
